docs/guides/nrfconnect_examples_software_update.md

```diff
@@ -1,15 +1,83 @@
-line tool.
+## Devide Firmware Upgrade over Matter
 
-## Device Firmware Upgrade using smartphone
+> **_NOTE:_** The procedure presented below requires that you have OpenThread
+> Border Router set up either in Docker or on a Raspberry Pi. Read
+> [Setup OpenThread Border Router on Raspberry Pi](openthread_border_router_pi.md)
+> to learn how to install the OTBR on a Raspberry Pi.
+
+The DFU over Matter involves two kinds of nodes: OTA Provider and OTA Requestor.
+The OTA Provider is a node that can respond to the OTA Requestors' queries about
+available software updates and share the update packages with them. The OTA
+Requestor is any node that needs to be updated and can communicate with the OTA
+Provider to fetch applicable software updates. In the procedure described below,
+the OTA Provider will be a Linux application and the example running on the
+Nordic Semiconductor's board will work as the OTA Requestor.
+
+To test the DFU over Matter, you need to complete the following steps:
+
+1.  Navigate to the CHIP root directory.
+2.  Build OTA Provider application for Linux:
+
+        $ scripts/examples/gn_build_example.sh examples/ota-provider-app/linux out/provider chip_config_network_layer_ble=false
+
+3.  Build chip-tool for Linux:
+
+        $ scripts/examples/gn_build_example.sh examples/chip-tool out/chiptool 'chip_mdns="platform"'
+
+4.  Run OTA Provider application with _app\_update.bin_ replaced with the path to
+    the new firmware image which you wish to provide to the Matter device:
+
+         $ out/provider/chip-ota-provider-app -f app_update.bin
+
+    Keep the application running and use another terminal for the remaining
+    steps.
+
+5.  Commission the OTA Provider into the Matter network using Node Id 1:
+
+        $ ./out/chiptool/chip-tool pairing onnetwork 1 20202021
+
+6.  Use the OTBR web interface to form a new Thread network using the default
+    network settings.
+7.  Commission the Matter device into the same Matter network using Node Id 2.
+    The parameter starting with _hex:_ prefix is the Thread network's Active
+    Operational Dataset. It can be retrieved from the OTBR in case you have
+    changed the default network settings when forming the network.
+
+         $ ./out/chiptool/chip-tool pairing ble-thread 2 hex:000300000f02081111111122222222051000112233445566778899aabbccddeeff01021234 20202021 3840
+
+8.  Initiate the DFU procedure in one of the following ways:
+
+    -   If you have built the device firmware with `-DCONFIG_CHIP_LIB_SHELL=y`
+        option, which enables Matter shell commands, run the following command
+        on the device shell. The numeric arguments are Fabic Index, Provider
+        Node Id and Provider Endpoint Id, respectively.
+
+               $ matter ota query 1 1 0
+
+    -   Otherwise, use chip-tool to send Announce OTA Provider command to the
+        device. Note that Provider Node Id and Provider Endpoint Id are sent in
```

Comment:
Provider or Requestor? If it's not mistake, I am confused why before example shows Node ID 1 and now 2.

Reply:
Right, I decided to explain all parameters instead.